packages/beacon-node/src/network/peers/discover.ts

```diff
@@ -245,18 +261,69 @@ export class PeerDiscovery {
     }
   }
 
+  /**
+   * Progressively called by libp2p peer discovery as a result of any query.
+   */
+  private onDiscoveredPeer = async (evt: CustomEvent<PeerInfo>): Promise<void> => {
```

Comment:
so now we have onDiscoveredPeer and onDiscoveredENR, could them be called together on the same peer? 

that's already an issue with v1.2.1
https://github.com/ChainSafe/lodestar/issues/4790